foundation/org.eclipse.persistence.core/src/main/java/org/eclipse/persistence/internal/helper/ConcurrencyUtil.java

```diff
@@ -0,0 +1,1490 @@
+                        currentReadLockAcquiredAndNeverReleased.getDateOfReadLockAcquisition(),
+                        currentReadLockAcquiredAndNeverReleased.getNumberOfReadersOnCacheKeyBeforeIncrementingByOne(),
+                        currentReadLockAcquiredAndNeverReleased.getCurrentThreadStackTraceInformationCpuTimeCostMs()}));
+                String stackTraceInformation = currentReadLockAcquiredAndNeverReleased.getCurrentThreadStackTraceInformation();
+                if (stackTraceStringToStackTraceExampleNumber.containsKey(stackTraceInformation)) {
+                    // we can spare the massive dump from being any fatter we have alreayd added a stack trace id that
+                    // is identical to the stack trace were were about dump
+                    // we just refer to the stack trace id.
+                    writer.write(TraceLocalization.buildMessage("concurrency_util_summary_read_locks_on_thread_step002_4", new Object[] {readLockNr, stackTraceStringToStackTraceExampleNumber.get(stackTraceInformation)}));
+                } else {
+                    // Since we have not see this stack trace pattern for this thread yet we will dump the stack trace
+                    // into the massive dump giving it a new global id
+                    long stackTraceId = stackTraceIdAtomicLong.incrementAndGet();
+                    stackTraceStringToStackTraceExampleNumber.put(stackTraceInformation, stackTraceId);
+                    writer.write(TraceLocalization.buildMessage("concurrency_util_summary_read_locks_on_thread_step002_5", new Object[] {readLockNr, stackTraceId, stackTraceInformation}));
+                }
+                writer.write("\n\n");
+            }
+        }
+
+        // (d) We have some more information to pump out namely errors we have traced each time the number of readers was decremented
+        writer.write("\n\n");
+        writer.write(TraceLocalization.buildMessage("concurrency_util_summary_read_locks_on_thread_step002_6", new Object[] {threadName, readLockManager.getRemoveReadLockProblemsDetected().size()}));
+        long releaseReadLockProblemNr = 0;
+        for (String currentReleaseReadLockProblem : readLockManager.getRemoveReadLockProblemsDetected()) {
+            releaseReadLockProblemNr++;
+            writer.write(TraceLocalization.buildMessage("concurrency_util_summary_read_locks_on_thread_step002_7", new Object[] {releaseReadLockProblemNr, currentReleaseReadLockProblem}));
+        }
+        writer.write("\n\n");
+        return writer.toString();
+    }
+
+    private static String getPropertyConcurrencyManagerAcquireWaitTime() {
+        return (PrivilegedAccessHelper.shouldUsePrivilegedAccess()) ?
+                AccessController.doPrivileged(new PrivilegedGetSystemProperty(SystemProperties.CONCURRENCY_MANAGER_ACQUIRE_WAIT_TIME))
+                : System.getProperty(SystemProperties.CONCURRENCY_MANAGER_ACQUIRE_WAIT_TIME, "0");
```

Comment:
Privileged call does not use default value, non-privileged does use one - intentional? Also note that [`Long.getLong()`](https://docs.oracle.com/javase/8/docs/api/java/lang/Long.html#getLong-java.lang.String-java.lang.Long-) could probably be used here as I see no handling of invalid values on this place.